src/hotspot/os/posix/perfMemory_posix.cpp

```diff
@@ -428,6 +428,23 @@ static bool is_file_secure(int fd, const char *filename) {
   return true;
 }
 
+// return the file name of the backing store file for the named
+// shared memory region for the given user name and vmid.
+//
+// the caller is expected to free the allocated memory.
+//
+static char* get_sharedmem_filename(const char* dirname, int vmid, int nspid) {
+
+  int pid = LINUX_ONLY((nspid == -1) ? vmid : nspid) NOT_LINUX(vmid);
+
+  // add 2 for the file separator and a null terminator.
+  size_t nbytes = strlen(dirname) + UINT_CHARS + 2;
+
+  char* name = NEW_C_HEAP_ARRAY(char, nbytes, mtInternal);
+  snprintf(name, nbytes, "%s/%d", dirname, pid);
+
+  return name;
+}
```

Comment:
Hello Johan, I'll take a look at the code if there are other places where this is happening. Since this is existing code (not added by this PR), if there are too many other unrelated places where this is happening then I'll create a separate issue to address it. Otherwise, I'll do it in this PR itself. Thank you for catching this.